File: app/src/main/java/com/michalbaran/incomeapp/IncomeViewAdapter.java
package com.michalbaran.incomeapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.michalbaran.incomeapp.database.Income;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.TimeZone;

public class IncomeViewAdapter extends RecyclerView.Adapter<IncomeViewAdapter.ViewHolder> {

    private Context context;
    private  List<Income> incomeList;

    public IncomeViewAdapter (Context context, List<Income> incomeList){
        this.context=context;
        this.incomeList=incomeList;
    }

    public static String formatDate(Date date) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("dd MMMM yyyy");
        dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
        return dateFormat.format(date);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(this.context).inflate(R.layout.income_recycler_row,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        final Income income = this.incomeList.get(position);
        holder.category.setText(income.getCategory());
        holder.date.setText(formatDate(income.getDate()));
        holder.amount.setText(income.getAmount().toString());
    }

    @Override
    public int getItemCount() {
        return this.incomeList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        private TextView category, date, amount;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            this.category = itemView.findViewById(R.id.txtRecyclerCategory);
            this.date = itemView.findViewById(R.id.txtRecyclerDate);
            this.amount = itemView.findViewById(R.id.txtRecyclerAmount);

        }
    }
}

File: app/src/main/java/com/michalbaran/incomeapp/IncomeViewAdapter.kt
package com.michalbaran.incomeapp

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.michalbaran.incomeapp.database.Income

import java.text.SimpleDateFormat
import java.util.Date
import java.util.TimeZone

class IncomeViewAdapter(private val context: Context, private val incomeList: List<Income>) : RecyclerView.Adapter<IncomeViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(this.context).inflate(R.layout.income_recycler_row, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val income = this.incomeList[position]
        holder.category.setText(income.getCategory())
        holder.date.text = formatDate(income.date)
        holder.amount.text = income.amount!!.toString()
    }

    override fun getItemCount(): Int {
        return this.incomeList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val category: TextView
        private val date: TextView
        private val amount: TextView


        init {
            this.category = itemView.findViewById(R.id.txtRecyclerCategory)
            this.date = itemView.findViewById(R.id.txtRecyclerDate)
            this.amount = itemView.findViewById(R.id.txtRecyclerAmount)

        }
    }

    companion object {

        fun formatDate(date: Date?): String {
            val dateFormat = SimpleDateFormat("dd MMMM yyyy")
            dateFormat.timeZone = TimeZone.getTimeZone("GMT")
            return dateFormat.format(date!!)
        }
    }
}
